util/validators.ts

```diff
@@ -29,3 +35,37 @@ export function normalizeString(rawString: unknown): string {
     } 
     throw new Error('Front-matter: Value must be a string.')
 }
+
+export function normalizeFrontMatter(rawFrontMatter: ReturnType<typeof greyMatter>['data']): FrontMatter {
+    return {
+        title: normalizeString(rawFrontMatter.title) ?? normalizeString(rawFrontMatter.alternateTitle) ?? 'Untitled Document',
+        alternateTitle: rawFrontMatter.alternateTitle ? normalizeString(rawFrontMatter.alternateTitle) : null,
+        tags: normalizeStringArray(rawFrontMatter.tags),
+        published: isBoolean(rawFrontMatter.published) ? rawFrontMatter.published : true,
+        unlisted: isBoolean(rawFrontMatter.unlisted) ? rawFrontMatter.unlisted : false,
+        author: rawFrontMatter.author ? normalizeString(rawFrontMatter.author) : null,
+        image: rawFrontMatter.image ? normalizeString(rawFrontMatter.image) : null,
+        imageAlt: rawFrontMatter.imageAlt ? normalizeString(rawFrontMatter.imageAlt) : null,
+        socialImage: rawFrontMatter.socialImage ? normalizeString(rawFrontMatter.socialImage) : null,
+        description: rawFrontMatter.description ? normalizeString(rawFrontMatter.description) : null,
+        type: normalizeString(rawFrontMatter.type)
+    }
+}
+
+export function normalizeRecordCollectionDefinition(collection: RecordCollectionDefinition): RecordCollectionDefinition {
+    return {
+        title: normalizeString(collection.title),
+        slug: normalizeString(collection.slug),
+        type: normalizeString(collection.type),
+        members: normalizeStringArray(collection.members)
+    }
+}
+
+export function normalizeAuthorCollectionDefinition(author: AuthorCollection): AuthorCollection {
```

Comment:
The normalize functions look good to me. Just a thought for the future: If we normalize each field, I wonder if Typescript would let us define this as:
```suggestion
export function normalizeAuthorCollectionDefinition(author: unknown): AuthorCollection {
```
... which would let us drop the `as` clause when we load it in:

https://github.com/sourcegraph/learn/blob/f7335858acfb8f33d2c5e951742fbf6f7b0bb9cb/util/loadAuthorCollections.ts#L25